### PR TITLE
feat: add functions to get neighboring tiles

### PR DESCRIPTION
**Pull request by LordMidas**
The utility of the `getNeighbors` function is obvious. The utility of the `getNeighbor` function is shown below:
```squirrel
// Objective:
// Get the first empty tile adjacent to myTile

// current workflow
local targetTile;
for (local i = 0; i < 6; i++)
{
	if (myTile.hasNextTile(i))
	{
		local nextTile = myTile.getNextTile(i);
		if (nextTile.IsEmpty)
		{
			targetTile = nextTile;
			break;
		}
	}
}

// new workflow:
local targetTile = ::MSU.Tile.getNeighbor(myTile, @(tile) tile.IsEmpty);
```

### CODE REVIEW

File: msu/utils/tile.nut [hunk 1]
```diff
@@ -12,4 +12,29 @@
...
+		{
+			if (_tile.hasNextTile(i))
+				ret.push(_tile.getNextTile(i));
+		}
+		return ret;
+	}
+
+	function getNeighbor( _tile, _function )
+	{
+		// _function( tile ) and returns boolean
```
Comment (Enduriel) [inline msu/utils/tile.nut:29]: this needs to be better, doesn't really explain anything and should be using `_tile` regardless. (can be removed though and that would be fine imo.

**Review by Enduriel — CHANGES_REQUESTED** [review]

File: msu/utils/tile.nut (hunk 1 above)
Comment (LordMidas) [reply msu/utils/tile.nut:29]: The function is correct. It does the following: "Give me the first neighbor of `_tile` for which `_function` returns true`.

Comment (Enduriel) [reply msu/utils/tile.nut:29]: I'm not saying the function is wrong, I'm saying the comment isn't useful

**Review by Enduriel — CHANGES_REQUESTED** [review]
The comments here are still not very useful and should be changed or removed

**Comment by LordMidas** [comment]
> The comments here are still not very useful and should be changed or removed

I don't know why you say that. Looking at the comment it becomes clear what the format of the passed `_function` should be. And looking at the code's body it becomes clear how that function's return value is used. Why does the comment need to be any more elaborate than that? The code is self-explanatory.

**Review by Enduriel — APPROVED** [review]

**Review by TaroEld — APPROVED** [review]

**Pull request merged**